### ISSUE TITLE
Question about maintainership of the project

### ISSUE THREAD

**Issue by Sigmanificient** [ridiculousfish/cdecl-blocks#43]
Hi,

I recently adopted the `cdecl` build package from the [nixpkgs repository](https://github.com/NixOS/nixpkgs).

It seems that the `cdecl` project isn't receiving many updates. Today, I found [this repository](https://github.com/paul-j-lucas/cdecl) on GitHub, which claims to be the official source for `cdecl` from version 3.0 onwards. However, I cannot find any evidence to support this claim, as the `cdecl` website still points here.

Are you aware of @paul-j-lucas's fork, and could you confirm whether it is official? I tried to contact him, but he has not responded and has automatically deleted the issue regarding this question. This behavior raises suspicions about the authenticity of his claim and seems unprofessional.

If the mentioned repository is not genuine, I would suggest getting in contact with him to resolve this confusion.

Clarification on this matter would greatly help me in maintaining `cdecl` on nixpkgs.

Thanks

**Comment by Sigmanificient** [comment]
@paul-j-lucas Please to not delete GitHub issue for not being bug reports / enhancement proposals, as they are the primarily way for discussion on a repository (which has discussions disabled like yours). This behavior only reinforce confusion and is destructive to the project records. I couldn't even know that you have replied by mail!

Your repository history [is directly forking](https://github.com/paul-j-lucas/cdecl/commits/master/?after=ea55934f72e51cb0545a457f3cc566a9f53d7e4d+6470) @ridiculousfish's project which led me to think of this source to the genuine version. In this is in that regard that i used the term "official".

Btw, are you the person that [spent about 3 years unsuccessfully trying to track down the authors of cdecl](https://www.reddit.com/r/programming/comments/a3pm6/cdecl_the_c_gibberish_to_english_translator_is/)?
With my own research, I could find [this project](https://upc.lbl.gov/docs/user/upcdecl.html) that seems to goes along what your [README](https://upc.lbl.gov/docs/user/upcdecl.html) say.

Sorry for the confusion, but once again, this could have been solved more easily within a single issue. It would have also help identifying the lack of source from the readme and avoid unnecessary back and forth.

**Comment by paul-j-lucas** [comment]
> Please to not delete GitHub issue for not being bug reports / enhancement proposals, as they are the primarily way for discussion on a repository (which has discussions disabled like yours).

It's my repository and I'll run it as I see fit.

> I couldn't even know that you have replied by mail!

Because...?  Why don't you check your e-mail?  I sent e-mail to edhyjox@gmail.com which is the addresses given in your GitHub profile.

> Your repository history shows that it is a direct fork of @ridiculousfish's project ...

I know.

> ... which led me to believe that this source might be the genuine version.

That's just a bad assumption on your part. If you bothered to read the [README](https://github.com/paul-j-lucas/cdecl/blob/master/README.md) (which you are kind of expected to read — that's why it's called a "README") or the [AUTHORS](https://github.com/paul-j-lucas/cdecl/blob/master/AUTHORS) file, you hopefully would have come to hold a different belief.

I specifically give cdecl's history, and even link to the "original" source code (parts [1](https://groups.google.com/g/comp.sources.unix/c/Y76scbXQQBk/m/MVrZZBG0nNwJ) & [2](https://groups.google.com/g/comp.sources.unix/c/yzWbI4agBE0/m/ddqzmuiEidwJ)) on `comp.sources.unix`.

> Btw, are you the person that spent about 3 years unsuccessfully trying to track down the authors of cdecl?

No.

> Sorry for the confusion, but this could have been resolved more easily within a single issue. 

The issue might never have occurred in the first place if you bothered to read the README and check your e-mail.

The bottom line is:

+ The "original" Graham Ross code, modified by Tony Hansen, was written in 1988.  It obviously doesn't support modern C or C++.  Additionally, it has never had semantic error checking, command-line autocompletion, and a myriad of other features.
+ ridiculousfish took Tony Hansen's code, added only the single feature to support Apple's Blocks syntax, and released it.  That's it.
+ I took ridiculousfish's code (which, again, is only Hansen's + the code to support blocks), and added support for modern C and C++ as well as many new features.  I _could_ have started from Hansen's code, but I figured I might as well support the blocks syntax.

Use whichever version you want.

**Comment by Sigmanificient** [comment]
The simple answer is, like many, i do not check my email as often as my GitHub account (i check it only once a day).
I will not go further since while issues *can be used to keep track of bugs, enhancements, or other requests* (from GitHub doc), there aren't for such debate.

I indeed read your readme, but I had to open a discussion anyway as the 2 links in your README goes to, hard to understand, *google groups* posts, which do not seems like a trustworthy source to me. Please stop being rude as this isn't helpful

**Comment by paul-j-lucas** [comment]
> The simple answer is, like many, i do not check my email as often as my GitHub account ...

Not my problem.  Your response was to assume that I was being "suspicious" and "unprofessional" — another bad assumption.

> I indeed read your readme, but I had to open a discussion anyway as the 2 links in it goes to, hard to understand, google groups posts, which do not seems like a trustworthy source to me.

They're _not_ Google Groups posts; they're `comp.sources.unix` posts on [Usenet](https://en.wikipedia.org/wiki/Usenet) that's been archived by Google.  Many years ago, Google created "Google Groups" and also interfaced it to Usenet.  Usenet has long since been dead, so Google is now (AFAIK) the sole remaining repository of Usenet posts.  The fact that you don't consider such a source trustworthy is also a bad assumption.

> Please stop being rude as this isn't helpful.

When you accuse me of being suspicious and unprofessional all because you don't check your e-mail frequently, then, yes, I tend to get a bit annoyed.  I'll stop being rude when you apologize.

> I was considering creating a second package for your version, but your attitude towards me made be disinterested.

First, the original cdecl and the blocks version are pretty much useless since, as I've said, neither supports modern C or C++.  IMHO, the only package any distro should have is mine since it supports modern C and C++.

Second, if you want to give your users a package of a very outdated version of cdecl and deny them a modern version, that's pretty much punishing your users because your feelings got hurt.

**Comment by Sigmanificient** [comment]
At the time i posted this message, your actions of deleting the GitHub issue that were asking a genuine question on the project, seems indeed, to be suspicious. This is the unfortunate impression that it gave me as i tried to reach you.
I already apologized for the oversight of your project being possibly not genuine.

However i need to once again, kindly ask you to stop arguing about *which frequency someone should read their email* as this is not the place to do so.

I considered the links you put on your readme non-trustworthy as, like i mention, are quite hard to understand from an outside perspective. Saying that it was a "bad assumption" is not helpful, however I am glad you took the time for clarifications.

I tried to get a *positive discussion* going on, about `cdecl` history and maintainership and I got, at least, an answer to my questions. There is no need to continue this discussion especially on the debate you are starting.

**Comment by paul-j-lucas** [comment]
> At the time i posted this message, your actions of deleting the GitHub issue that were asking a genuine question on the project, seems indeed, to be suspicious.

And if you had only checked your e-mail **_before_** making any assumptions, this could have been avoided.

> I already apologized for the oversight of your project being possibly not genuine.

No; you wrote:

> Sorry for the confusion ...

So you apologized _only_ for the confusion, _not_ for impugning my integrity.

> However i need to once again, kindly ask you to stop arguing about which frequency someone should read their email as this is not the place to do so.

I'm **_NOT_** arguing that.  I couldn't care less how often you check your e-mail.  My problem is that — within _minutes_ — you concluded suspicion and unprofessionalism.  It's far more customary to give someone the benefit of the doubt and wait at least 24 hours before coming to any conclusions.  Had you waited, you likely _would_ have checked your e-mail in that time and gotten your answers.

> I considered the links you put on your readme non-trustworthy as, like i mention, are quite hard to understand from an outside perspective.  Saying that it was a "bad assumption" is not helpful ...

The hope was you would learn from this and stop making such bad assumptions in the future.  Apparently not.

**Comment by Sigmanificient** [comment]
As i said, I gave my honest **impression** of what it **seemed** (and absolutely not a conclusion), from the point of view of a package manager that ask a question which gets deleted. I explained above the need of careful considerations for a project to be the source of a package and why your actions can easily get interpreted as a red flag in this context. My intent is not to cause you problems and this is why i immediately reach for **clarification** and not attack.

Please, once again, stop arguing about emails, response time, etc.

**Comment by Sigmanificient** [comment]
Lets stay positive.

So if you feel that i impugned you integrity, sorry about that, it wasn't my goal.
Now, it would be nice for us to remove this entire side discussion (except the first few comments) to keep a clean record of the issue

**Comment by paul-j-lucas** [comment]
> As i said, I gave my honest impression of what it seemed (and absolutely not a conclusion) ...

My honest impression of you is that it "seems" that you are impatient, impudent, know little of Internet history, and, consequently, probably shouldn't be a package maintainer at all.

But please do not be offended by what I just wrote!  It's not a "conclusion" — it only _seems_ to be the case.  (Hopefully, you now understand my point here: using the word "seems" rather than "concludes" doesn't make what you wrote any less insulting.)

> Even though i am not a native speaker, you can see that the words i used are not trying to discredit you, so i do not understand why you keep going back to this.

Because:

1. You aren't acknowledging that your assumptions were bad.
2. You _choose_ to check e-mail infrequently while not understanding that such a choice has consequences.

> Please, once again, stop arguing about emails check frequency.

Again, I'm not.  Again, I'm saying that such a choice — which you're completely free and valid to make — has consequences and you should be cognizant of such consequences.

> So if you feel that i impugned you integrity, sorry about that, it wasn't my goal.

Finally, an actual apology.  Thanks.

> Now, it would be nice for us to remove this entire side discussion (except the first few comments) to keep a clean record of the issue. I will wait for a day to see if you cleaned the off-topic comments and will follow.

Fine with me, but you delete your comments first since you impugned me and I'm was merely defending myself.

**Comment by Sigmanificient** [comment]
hope you will follow
